rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -3257,6 +3251,24 @@ private boolean isXmlWorkflow(InputPart fileInputPart) {
                 .contains(MediaType.APPLICATION_XML.toLowerCase());
     }
 
+    protected static Map<String, String> createSortableTaskAttrMap() {
+        HashMap<String, String> sortableTaskAttrMap = new HashMap<>(13);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "sortableTaskAttrMap" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)